kratos/factories/base_factory.h

```diff
@@ -0,0 +1,236 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+#if !defined(KRATOS_BASE_FACTORY_H_INCLUDED )
+#define  KRATOS_BASE_FACTORY_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "includes/kratos_components.h"
+#include "includes/shared_pointers.h"
+
+namespace Kratos
+{
+///@name Type Definitions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class FactoryMethods
+ * @ingroup KratosCore
+ * @brief Here we define some common methods
+ * @details Defines the base class factory methods
+ * @author Vicente Mataix Ferrandiz
+ */
+class FactoryMethods
```

Comment:
can you please explain why this is needed?

Reply:
In order to have a base class not templated in order to expose to python methods, reducing the code duplication

Reply:
ok that makes sense